mne_bids/utils.py

```diff
@@ -405,9 +411,16 @@ def _read_events(events_data, raw):
             raise ValueError('Events must have second dimension of length 3, '
                              'found %s' % events_data.shape[1])
         events = events_data
-    else:
+    elif 'stim' in raw:
         events = find_events(raw, min_duration=0.001, initial_event=True)
-    return events
+    elif ext in ['.vhdr', '.set'] and check_version('mne', '0.18.dev0'):
```

Comment:
isn't reading from annotations also supported for `.edf` data? Furthermore, `.fif` seems to be supported ... but `.fif` will always have a stim channel, so this part of the if-else-clause will never be reached for `.fif`.

reference:
https://github.com/mne-tools/mne-python/blob/d7fe74851707950408d2c2b2e34dfa5fdc1d0669/mne/annotations.py#L510-L514

Reply:
For .fif' you already have stim channel, so that's where it enters the if-else clause. I think annotations in .fif are not the same as in .set. As far as my understanding goes, in .fif files annotations have a duration but not necessarily in .set or .vhdr files -- so they are events stored in the annotation object (I may be wrong though ...) 

Now, edf is more complicated -- so I had left it out. I am not exactly sure what happens. There are TAL channels which store the annotations with durations. But I'm not sure where the events are stored. I think I need @agramfort's or @massich's expertise for this :)

Reply:
Sorry for my late response. **EDF** has no notion of stim channel. They use the **TAL** channel instead. The **TAL** channel, is a collection of the following triplests: `(start time, duration, description)`, which naturally translate into `mne.Annotaiton`. **EDF+** the TAL channel is mandatory with at very least **start recording** and  **end recording** (the names are not really those, we should check the standard). Meanwhile I think that **EDF** the **TAL** is not mandatory, but its there in most of our examples. 

Once said that, we have EDF files with no **TAL** (or with only start stop) where the stim channel has been recorded in one of the channels.

But either way, there is no notion of `events` (a matrix of latencies where things happen during recording). `.bdf` has it though. The issue here is that **mne** interpretes `.bdf` as if it were an **EDF**, therefore, we have no access to such a matrix from the public API and we encode this event table into the Annotations as if we had read an **EDF** file.

Reply:
Perfect, thank you for the clear explanation @massich . 

So, I guess we need to add .bdf to this list @sappelhoff. But I'm not losing hair over it yet until we have a user who comes and complains ;)

Reply:
> So, I guess we need to add .bdf to this list

and .edf as well, if I understood correctly 

> But I'm not losing hair over it yet until we have a user who comes and complains ;)

yes, that's right :-)